pkg/backend/display/progress.go

```diff
@@ -619,8 +622,17 @@ func (display *ProgressDisplay) printDiagnostics() bool {
 				display.println("")
 			}
 		}
+	}
 
+	// Print a link to Copilot to explain the failure.
+	// Check for SuppressPermalink ensures we don't print the link for DIY backends
+	if wroteDiagnosticHeader && !display.opts.SuppressPermalink && !display.opts.SuppressLinkToCopilot {
+		display.println("    " +
+			colors.SpecCreateReplacement + "[Pulumi Copilot]" + colors.Reset + " Would you like help with these diagnostics?")
```

Comment:
Diagnostics are emitted for a fairly wide variety of reasons - can we more precisely catch when an update fails instead?

Reply:
This should work for any failure, the Copilot should be able to explain it based on the context in the permalink.